src/main/java/de/danielbechler/diff/node/DiffNode.java

```diff
@@ -732,6 +740,16 @@ else if (childCount() > 1)
 		return sb.toString();
 	}
 
+	private Set<String> getCategoriesFromConfig() {
+
+		return categoriesFromConfig;
+	}
+
+	public void setCategoriesFromConfig(Set<String> categoriesFromConfig) {
+
+		this.categoriesFromConfig = categoriesFromConfig;
+	}
```

Comment:
Unfortunately there is no way to avoid making this method part of the public API, so people will eventually start (ab)using it. That's one of the reasons why I'd prefer a different name for the field, as mentioned above. Although I wouldn't even expose the field via getter or setter at all.

If we can't avoid leaking this detail to the outside, let's make it a cool feature: let's say we call the method `addCategories` and instead of replacing the categories, we just append them to a set. Then people are able to add their own categories later on, when traversing the result. That may allow for some advanced filtering based on criteria that go far beyond the things the ObjectDiffer is (or should be) capable of.

Calling it `addCategories` suggests the direct relationship with `getCategories` and by making it append-only people won't be confused when they try to remove a category that comes from the annotation or the parent node.

I imagine something like this:

```java
public final void addCategories(final Collection<String> additionalCategories)
{
	Assert.notNull(additionalCategories, "additionalCategories");
	this.additionalCategories.addAll(additionalCategories);
}
```

In this example I changed the parameter type from `Set` to `Collection`, because the underlying object is a `Set` anyway so we don't care what collection type they pass.